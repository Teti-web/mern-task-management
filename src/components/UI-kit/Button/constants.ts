export enum IconStyleButton {
  ICON_LEFT = "icon-left",
  ICON_RIGHT = "icon-right",
  ICON_ONLY = "icon-only",
}
