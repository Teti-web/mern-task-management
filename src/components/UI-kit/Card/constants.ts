export const CARD_TEST_ID = "cardComponent";
