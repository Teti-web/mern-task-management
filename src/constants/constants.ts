export enum Size {
  SMALL = "small",
  MEDIUM = "medium",
  LARGE = "large",
  CIRCLE = "circle",
}

export enum Style {
  PRIMARY = "primary",
  SECONDARY = "secondary",
  MINIMAL = "minimal",
  CIRCLE = "circle",
}
