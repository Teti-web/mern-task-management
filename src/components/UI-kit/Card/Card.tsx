import { FC } from "react";
import { ClockIcon } from "@heroicons/react/24/outline";

import ProgressBar from "../ProgressBar/ProgressBar";
import { CARD_TEST_ID } from "./constants";

export type CardType = {
  img: string;
  altImage?: string;
  title: string;
  suptitle: string;
  timeCreated: string;
  progressValue: number; //min 0, max 100
};

const Card: FC<CardType> = ({
  img,
  altImage,
  title,
  suptitle,
  timeCreated,
  progressValue,
}) => {
  return (
    <div
      className="bg-white rounded-[10px] p-[20px] md:p-[24px] grid grid-cols-1 gap-[16px] max-w-[340px]"
      data-testid={CARD_TEST_ID}
    >
      <img
        className="rounded-[10px] object-center object-cover w-full h-[110px]"
        src={img}
        alt={altImage}
      />
      <div className="font-primary">
        <h3 className="text-secondinary-500 font-semibold text-left text-[16px]">
          {title}
        </h3>
        <p className="text-[12px] font-medium text-left text-secondinary-400 pt-[5px]">
          {suptitle}
        </p>
      </div>
      <div className="flex flex-col gap-[8px]">
        <div className="flex justify-between font-primary font-medium text-[16px]">
          <p className="text-secondinary-500">Process</p>
          <p className="text-primary-500">{progressValue}%</p>
        </div>
        <ProgressBar value={progressValue} />
      </div>
      <div className="flex flex-row gap-[5px]">
        <ClockIcon className="w-[24px] h-[24px] text-secondinary-400" />
        <p className="font-primary font-medium text-[16px] text-secondinary-500">
          {timeCreated}
        </p>
      </div>
    </div>
  );
};

export default Card;
