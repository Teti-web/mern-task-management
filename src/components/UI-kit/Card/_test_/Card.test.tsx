import "@testing-library/jest-dom";
import { render, cleanup } from "@testing-library/react";
import Card, { CardType } from "../Card";
import { CARD_TEST_ID } from "../constants";

const mockCardData: CardType = {
  img: "mock-image-url",
  altImage: "Mock Alt Image",
  title: "Mock Title",
  suptitle: "Mock Subtitle",
  timeCreated: "Mock Time",
  progressValue: 50,
};

describe("Card componet", () => {
  afterEach(cleanup);

  it("renders with correct data", () => {
    const { getByTestId, getByAltText, getByText, asFragment } = render(
      <Card {...mockCardData} />
    );

    expect(getByTestId(CARD_TEST_ID)).toBeInTheDocument();
    expect(getByAltText("Mock Alt Image")).toBeInTheDocument();
    expect(getByText("Mock Title")).toBeInTheDocument();
    expect(getByText("Mock Subtitle")).toBeInTheDocument();
    expect(getByText("50%")).toBeInTheDocument();
    expect(getByText("Mock Time")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
