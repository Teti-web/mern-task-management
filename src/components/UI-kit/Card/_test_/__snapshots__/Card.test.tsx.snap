// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card componet renders with correct data 1`] = `
<DocumentFragment>
  <div
    class="bg-white rounded-[10px] p-[20px] md:p-[24px] grid grid-cols-1 gap-[16px] max-w-[340px]"
    data-testid="cardComponent"
  >
    <img
      alt="Mock Alt Image"
      class="rounded-[10px] object-center object-cover w-full h-[110px]"
      src="mock-image-url"
    />
    <div
      class="font-primary"
    >
      <h3
        class="text-secondinary-500 font-semibold text-left text-[16px]"
      >
        Mock Title
      </h3>
      <p
        class="text-[12px] font-medium text-left text-secondinary-400 pt-[5px]"
      >
        Mock Subtitle
      </p>
    </div>
    <div
      class="flex flex-col gap-[8px]"
    >
      <div
        class="flex justify-between font-primary font-medium text-[16px]"
      >
        <p
          class="text-secondinary-500"
        >
          Process
        </p>
        <p
          class="text-primary-500"
        >
          50%
        </p>
      </div>
      <div
        class="bg-primary-200 rounded-[8px] w-full h-[8px]"
      >
        <div
          class="bg-primary-500 rounded-[8px] h-full relative
         after:rounded-full after:bg-primary-500 after:block after:border-[2px] after:border-white after:border-solid after:w-[16px] after:h-[16px] after:absolute after:right-0 after:top-1/2 after:-translate-y-1/2"
          style="width: 50%;"
        />
      </div>
    </div>
    <div
      class="flex flex-row gap-[5px]"
    >
      <svg
        aria-hidden="true"
        class="w-[24px] h-[24px] text-secondinary-400"
        data-slot="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 6v6h4.5m4.5 0a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <p
        class="font-primary font-medium text-[16px] text-secondinary-500"
      >
        Mock Time
      </p>
    </div>
  </div>
</DocumentFragment>
`;
